lib/mpl_toolkits/tests/test_axes_grid.py

```diff
@@ -43,6 +44,9 @@ def test_imagegrid_cbar_mode_edge():
         # "second" ones.  To achieve this, clear out the axes first.
         for ax in grid:
             ax.cax.cla()
+            # We don't want to share the norms between the colorbar axes
+            # so create a deepcopy for each image
+            ax.images[0].norm = copy.deepcopy(ax.images[0].norm)
```

Comment:
Did we used to rely on a race condition in drawing to make this test work?

I'm also not seeing in the above code how they ended up coupled.

Reply:
This one took me a long time to figure out that this one-line fix is all that is needed. If I remove the deepcopy, the scales on the colorbar axes seem to stick around or not get updated appropriately? There are some extra ticks floating around and some spines don't get fully removed. I think there is some state that isn't fully getting cleared in `axes_grid` with the way the colorbars are handled there.

I don't think this is the most ideal test with this comment about clearing and overwriting portions of the axes.
```python
        # In each row/column, the "first" colorbars must be overwritten by the
        # "second" ones.  To achieve this, clear out the axes first.
```

Reply:
This test bit me again... I just added some new code to connect/disconnect the callbacks when changing the norm on a ScalarMappable that already exists. Now, I'm able to remove the `ax.cax.cla()` call, but I need a brand new ScalarMappable for only this test. I think there is something with the colorbars trying to modify old state here, which is why creating a new instance that has no sharing is what we need.

I still don't 100% follow this test code and haven't taken the time to really investigate it in depth and determine whether this test could be rewritten, or if something else in colorbars should be updated.